src/main/groovy/edu/oregonstate/mist/locations/frontend/resources/LocationResource.groovy

```diff
@@ -0,0 +1,261 @@
+
+import com.codahale.metrics.annotation.Timed
+import com.fasterxml.jackson.databind.ObjectMapper
+import edu.oregonstate.mist.api.AuthenticatedUser
+import io.dropwizard.auth.Auth
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.WebApplicationException
+import javax.ws.rs.core.Context
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.MultivaluedMap
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.UriInfo
+import java.util.regex.Pattern
+
+@Path("/locations")
+@Produces(MediaType.APPLICATION_JSON)
+class LocationResource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocationResource.class)
+    public static final ArrayList<String> ALLOWED_CAMPUSES = ["corvallis"]
+    public static final ArrayList<String> ALLOWED_TYPES = ["building", "dining"]
+
+    /**
+     * Default page number used in pagination
+     * @todo: should this be coming from skeleton or configuration file?
+     */
+    public static final Integer DEFAULT_PAGE_NUMBER = 1
+
+    /**
+     * Default page size used in pagination
+     * @todo: should this be coming from skeleton or configuration file?
+     */
+    public static final Integer DEFAULT_PAGE_SIZE = 10
+
+    @Context
+    UriInfo uriInfo
+
+    private final Map<String, String> locationConfiguration
+
+    private static final Pattern illegalCharacterPattern = Pattern.compile(
+            '''(?x)       # this extended regex defines
+               (?!        # any character that is not
+                  [
+                   a-zA-Z # a letter,
+                   0-9    # a number,
+                   -      # a hyphen,
+                   _      # an underscore,
+                   \\.    # a period, or
+                   @      # an at sign
+                  ])
+               .          # to be an illegal character.
+            ''')
+
+    LocationResource(Map<String, String> locationConfiguration) {
+        this.locationConfiguration = locationConfiguration
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Timed
+    Response list(@QueryParam('q') String q, @QueryParam('campus') String campus, @QueryParam('type') String type,
+                  @Auth AuthenticatedUser authenticatedUser) {
+
+        def pageNumber = getPageNumber()
+        def pageSize = getPageSize()
+        def trimmedQ = sanitize(q?.trim())
+        def trimmedCampus = sanitize(campus?.trim())
+        def trimmedType = sanitize(type?.trim())
+
+        if (trimmedCampus && !ALLOWED_CAMPUSES.contains(trimmedCampus)) {
+            throw new WebApplicationException(Response.Status.NOT_FOUND)
+        }
+
+        if (trimmedType && !ALLOWED_TYPES.contains(trimmedType)) {
+            throw new WebApplicationException(Response.Status.NOT_FOUND)
+        }
```

Comment:
see comment in previous pull request regarding `edu.oregonstate.mist.api.Resource.notFound()`